docs/01-Headless/03-Directives.md

```diff
@@ -1,5 +1,120 @@
+
+- `update` function, called when the context changed and you need either to update the directive configuration or to re-execute partially the original function or modify some internal state of the directive.
+- `destroy` function, used to remove eventual listeners, subscriptions or any reactive objects created by the directive. This function is usually called when the DOM element on which you applied the directive gets removed from the DOM.
+
+```typescript
+const createMyDirective: Directive<string> = (element: HTMLElement, text: string) => {
+	console.log('Directive has been executed on node ', element.id);
+	const clickListener = (event: Event) => {
+		console.log(text, event.target);
+	};
+	element.addEventListener('click', clickListener);
+	return {
+		update: (newText) => (text = newText),
+		destroy: () => {
+			node.removeEventListener('click', clickListener);
+			console.log('Destroy function executed ', element.id);
+		},
+	};
+};
+```
+
+This example shows a very simple directive that creates an event listener on `click` events that happen on the HTML element and then print the parameter `text` in the console, along with the clicked target.
+
+Both `update` and `destroy` functions are provided by the directive, as following:
+
+- `update`: gives a way to modify the original text to something new, when the context changes (see in the next section).
+- `destroy`: remove the previously created listener, so that you don't bloat the main thread with unused listeners.
+
+### How to use a directive
+
+Considering this HTML page, in which you have a `div` container `focus-element` that includes 2 `button`, an input text `clickText` and another `button` that removes the container.
+
+```html
+<div id="content">
+	<div id="focus-element">
+		<button id="test">button 1</button>
+		<button id="test2">button 2</button>
+	</div>
+	<hr />
+	<input id="clickText" type="text" />
+	<hr />
+	<button onclick="document.getElementById('focus-element').remove()">remove dom element</button>
+</div>
+```
+
+To use the directive in vanilla TS you need the following steps:
+
+1. Create the directive on a DOM element. In this case, our elementwill be the `focus-element` container.
+
+```typescript
+const trackElement = document.getElementById('focus-element');
+const focusElementDirective = createMyDirective(trackElement, 'focus-element click');
+```
+
+By doing it, when you click on a `button` inside the `focus-element`, you will get a log into your console,
+
+```
+focus-element click
+HTMLButtonElement{attributes: {…}, innerHTML: "button 1", nodeType: 1, tagName: "button"}
+```
+
+2. Observe context changes and call the `update` function accordingly. For example, we want to change the text to the value of the `clickText` input when it changes.
+
+```typescript
+const input = document.getElementById('clickText');
+input.addEventListener('change', (event) => {
+	instance && instance.update(event.target.value);
```

Comment:
Is it `focusElementDirective` instead of `instance` ?

It makes me think that it would be better, in my opinion, to have a running example here, even if the code is repeated for explanations (maybe they can also be included in the comments).

The question for the example is : where to put it, as it does not belong to any framework ?